example/extensions/lib_custom_op/README.md

```diff
@@ -146,21 +155,154 @@ Let’s take a closer look at those registry functions:
+        int64_t N = inputs[0].size();
+        int block = 256;
+        int grid = (N + (block - 1)) / block;
+        my_op_gpu_forward<<<grid,block,0,cuda_stream>>>(out_data, in_data, N);
+        return MX_SUCCESS;
+    }
+```
+
+Note that the `cuda_stream` object used for launching kernels is passed from MXNet backend via `OpResource` object. See below for details of `Operator Resource`.
+
+The CUDA kernel function should be no difference with an ordinary CUDA function.
+```c++
+    __global__ void my_op_gpu_forward(float *out, float *in, int64_t N) {
+        int tid = blockIdx.x * blockDim.x + threadIdx.x;
+        if (tid < N) {
+            // code your computational logic here
+        }
+    }
+```
+
+### Stateful GPU Custom Operator
+
+Recall that for writing custom stateful operator, you need to define a class that inherits `CustomStatefulOp` and overrides the `forward` and `backward` functions. Since stateful operator creating is context-aware, you need to create different classes for GPU and CPU stateful operators separately. the registering function will look like this:
+```c++
+    REGISTER_OP(my_state_op_gpu)
+    ...
+    .setCreateOpState(createOpStateCPU, "cpu")
+    .setCreateOpState(createOpStateGPU, "gpu");
+```
+
+Then you can create different classes for CPU and GPU stateful operators. MXNet will create the stateful operator instance based on the running context when the operator is invoked, and call stateful `forward` or `backward` function from the instantiated stateful operator class.
+```c++
+    class MyStatefulOpCPU : public CustomStatefulOp {
+    public:
+        explicit MyStatefulOpCPU() {}
+        MXReturnValue Forward(...) {
+            // code your CPU forward computational logic here
+        }
+        MXReturnValue Backward(...) {
+            // code your CPU backward computational logic here
+        }
+        ~MyStatefulOpCPU() {}
+    };
+
+    class MyStatefulOpGPU : public CustomStatefulOp {
+    public:
+        explicit MyStatefulOpGPU() {}
+        MXReturnValue Forward(...) {
+            // code your GPU forward computational logic here
+        }
+        MXReturnValue Backward(...) {
+            // code your GPU backward computational logic here
+        }
+        ~MyStatefulOpGPU() {}
+    };
+
+    MXReturnValue createOpStateCPU(std::map<std::string,std::string> attrs,
+                                   CustomStatefulOp** op_inst) {
+        *op_inst = new MyStatefulOpCPU();
+        return MX_SUCCESS;
+    }
+
+    MXReturnValue createOpStateGPU(std::map<std::string,std::string> attrs,
+                                   CustomStatefulOp** op_inst) {
+        *op_inst = new MyStatefulOpGPU();
+        return MX_SUCCESS;
+    }
+```
+
+Optionally, you can use the same class for CPU and GPU, but you’ll need to check the `MXContext` type in the `MXTensors` to dispatch CPU or GPU `forward` or `backward` functions yourself to do the computation.
+
+### Operator Resource
```

Comment:
Lets make this a ## section, so its not under the GPU section only since it applies to CPU too